services/provider/proto/provider.proto

```diff
@@ -143,6 +143,8 @@ message GetDesiredClientStateResponse{
     // kubeObjects holds the definition and metadata for all kube objects to be reconciled
     // by the client operator
     repeated KubeObject kubeObjects = 6;
+    // NonResilientPoolsTopologyKey for replica-1
+    string nonResilientPoolsTopologyKey = 7;
```

Comment:
Just confirming, there will never be a need to distinguish default empty & explicit empty string for this field?

Reply:
Yes, it can never be `explicit empty ` as if failure domain not exist the storagecluster reconcile will not complete.
PLus there is no difference